dev/deps/spark-deps-hadoop-2.3

```diff
@@ -73,32 +74,41 @@ hadoop-yarn-client-2.3.0.jar
 hadoop-yarn-common-2.3.0.jar
 hadoop-yarn-server-common-2.3.0.jar
 hadoop-yarn-server-web-proxy-2.3.0.jar
+hk2-api-2.4.0-b34.jar
+hk2-locator-2.4.0-b34.jar
+hk2-utils-2.4.0-b34.jar
 httpclient-4.3.2.jar
 httpcore-4.3.2.jar
 ivy-2.4.0.jar
 jackson-annotations-2.5.3.jar
 jackson-core-2.5.3.jar
 jackson-core-asl-1.9.13.jar
 jackson-databind-2.5.3.jar
-jackson-jaxrs-1.9.13.jar
+jackson-jaxrs-base-2.5.3.jar
+jackson-jaxrs-json-provider-2.5.3.jar
 jackson-mapper-asl-1.9.13.jar
+jackson-module-jaxb-annotations-2.5.3.jar
 jackson-module-scala_2.11-2.5.3.jar
-jackson-xc-1.9.13.jar
 janino-2.7.8.jar
 java-xmlbuilder-1.0.jar
+javassist-3.18.1-GA.jar
+javax.annotation-api-1.2.jar
 javax.inject-1.jar
+javax.inject-2.4.0-b34.jar
```

Comment:
I wouldn't say problematic, but both provide the exact same set of classes, so it's at least redundant. Being a javax API, I at least assume the newer one is backwards compatible.